File: backend/src/main/java/com/se114p12/backend/domains/order/Order.java
package com.se114p12.backend.domains.order;

import com.se114p12.backend.domains.authentication.User;
import com.se114p12.backend.domains.shipper.Shipper;
import com.se114p12.backend.enums.OrderStatus;
import com.se114p12.backend.domains.BaseEntity;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.Instant;
import java.util.List;

@Entity
@Table(name = "orders")
@Data
public class Order extends BaseEntity {

  @NotNull
  private Instant orderDate;

  @NotNull
  private String shippingAddress;

  private Long totalPrice;

  private String note;

  @NotNull
  private Instant expectedDeliveryTime;

  @NotNull
  private Instant actualDeliveryTime;

  @Enumerated(EnumType.STRING)
  private OrderStatus orderStatus;

//  private Payment paymentMethod;

  @ManyToOne
  @JoinColumn(name = "user_id")
  private User user;

  @ManyToOne
  @JoinColumn(name = "shipper_id")
  private Shipper shipper;

  @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
  private List<OrderDetail> orderDetails;
}

File: frontend/MAM/app/src/main/java/com/example/mam/entity/Order.kt
package com.example.mam.entity

import java.text.DecimalFormat
import java.time.Instant

data class Order(
    val id: String = "",
    val usedId: String = "",
    val orderDate: Instant ?= null,
    val paymentId: String = "",
    val shippingAddress: String = "",
    val orderItems: MutableList<OrderItem> =  mutableListOf(),
    var totalPrice: Int = 0,
    val note: String = "",
    val orderStatus: Int = 0,
    val expectDeliveryTime: Instant ?= null,
    val actualDeliveryTime: Instant?= null,
    val shipperId: String ?= null,
){
    fun getTotalToString(): String{
        calculateTotal()
        val formatter = DecimalFormat("#,###")
        return "${formatter.format(totalPrice)} VND"
    }
    fun calculateTotal() {
        totalPrice = orderItems.sumOf { it.price}
    }
}
